packages/w3up-client/src/client.js

```diff
@@ -291,9 +292,66 @@ export class Client extends Base {
         )
       }
     }
+
     return space
   }
 
+  /**
+   * Authorizes a gateway to serve content from the provided space and record egress events.
+   * Delegates the following capabilities to the gateway:
+   * - `space/content/serve/*`
+   *
+   * @param {import('./types.js').OwnedSpace} space - The space to authorize the gateway for.
+   * @param {object} [options] - Options for the authorization.
+   * @param {`did:web:${string}`} [options.gateway] - The Web DID of the gateway to authorize. If not provided, the default `did:web:w3s.link` gateway will be authorized.
+   * @param {number} [options.expiration] - The time in seconds to expire the authorization.
+   * @returns {Promise<import('./delegation.js').AgentDelegation<any>>} Resolves with the AgentDelegation instance once the gateway is successfully authorized.
+   */
+  async authorizeGateway(space, options = {}) {
```

Comment:
Would also be nice to not tie this to gateways specifically, since in the future we'll want to authorize bitswap nodes.

I'd suggest something like `authorizeContentServe` and instead of `gateway` as the option name perhaps `audience`.

When you create a space I'd have a more specific option like `{ authorizeGateways: ['did:web:w3s.link'], authorizePeers: ['did:web:elastic.dag.house'] }`

Reply:
Hmm. This is a bit tricky because we need to both authorize the server _and_ send that server the delegation. We can know how to speak to our own gateway, but speaking to arbitrary services gets trickier.

That said, perhaps this is a great use case for a [DID service endpoint](https://www.w3.org/TR/did-core/#dfn-serviceendpoint)! If each DID is responsible for specifying its own UCAN endpoint, we can simply talk to that. That also means everywhere in our system we specify a DID and a URL next to each other, we can stop doing that.

That's not a small thing to do—it'll take a fair bit of specification—but it might be a good ideal to point towards. In the meantime, I'd recommend making explicit, separate options for Freeway and Hoverboard in the client, which can become aliases for something more general later on.

Reply:
@alanshaw @Peeja For now, I think we can pass the `connection` option in the function arguments and assume that all the specified DIDs will be authorized, with their delegations stored in the provided service (the one associated with the connection). In the future, we can explore the idea of leveraging the DID service endpoint for this. WDYT?

Reply:
Love it! So the CLI will know about Freeway and Hoverboard (at least in the first iteration), but it stays abstract in the client library. I think that's a great way to go.